crates/nix_rs/src/copy.rs

```diff
@@ -0,0 +1,37 @@
+use crate::command::{CommandError, NixCmd};
+
+/// Runs `nix copy` in Rust
+pub async fn run_nix_copy(
+    cmd: &NixCmd,
+    host: &str,
+    omnix_input: &str,
+    path: &str,
+    extra_args: &[String],
```

Comment:
Per https://en.wikipedia.org/wiki/Separation_of_concerns you would delete these arguments and put a singular `Vec<PathBuf>` argument in their place.

Reply:
If this <https://github.com/juspay/omnix/pull/212#discussion_r1711652998> is confirmed, we could take additional `nix_args` as well, and the caller would only pass `--override-inputs` in it.